lib/rules/func-style.js

```diff
@@ -46,13 +55,22 @@ module.exports = {
         const style = context.options[0],
             allowArrowFunctions = context.options[1] && context.options[1].allowArrowFunctions,
             enforceDeclarations = (style === "declaration"),
+            exportFunctionStyle = context.options[1] && context.options[1].overrides && context.options[1].overrides.namedExports,
             stack = [];
 
         const nodesToCheck = {
             FunctionDeclaration(node) {
                 stack.push(false);
 
-                if (!enforceDeclarations && node.parent.type !== "ExportDefaultDeclaration") {
+                if (
+                    !enforceDeclarations &&
+                    node.parent.type !== "ExportDefaultDeclaration" &&
+                    typeof exportFunctionStyle === "undefined"
+                ) {
+                    context.report({ node, messageId: "expression" });
+                }
```

Comment:
This would be a false negative:

```js
/*eslint func-style: ["error", "expression", { "overrides": { "namedExports": "declaration" } }]*/

function foo() { // false negative
    // ...
}
```